### ISSUE TITLE
Integration tests are not always pulling head commit

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental-toolkit#1993]
I can't really find what is going on but I got a couple of clear examples that integration tests are not running the expected code. Most obvious is the difference from these two runs from PR #1992:

1. https://github.com/rancher/elemental-toolkit/actions/runs/8175584404
2. https://github.com/rancher/elemental-toolkit/actions/runs/8176020420

The first run and the second run they only diverge by this empty commit https://github.com/rancher/elemental-toolkit/pull/1992/commits/406899a016eb6f416e50ccc163bfacf4b5a713b0 on first run on workflow logs we can see the extra argument in makefile was not added in actual worlflow run, however we can see it in the second. In both cases the code was exactly the same.

I have see this issue more than once lately, but now I managed to collect an example that clearly shows the issue.

And the very same thing happened again on the same PR. Added a fix for recovery test

1. https://github.com/rancher/elemental-toolkit/actions/runs/8176968369 in this run we can see the recovery test does not run the modified command
2. https://github.com/rancher/elemental-toolkit/actions/runs/8177628804 shows that after another empty commit now runs the modified command as it was expected from the previous run.

Feels like if we were always testing one commit behind... so weird 🤷🏽‍♂️ This already puzzled me several times in the past few weeks...

**Comment by davidcassany** [comment]
Just found another example https://github.com/rancher/elemental-toolkit/actions/runs/8186424112?pr=1995

There is a cache hit, but the PR actually changes the code. Hence the cache hit is not possible as this hash includes tracking changes under `/pkg`, hence my guess is that this is just using a different code.

In fact in that same run if we check the toolkit commit is actually building this is the HEAD of `main` branch, hence the toolkit built with this tests is not including the actual changes. Gonna append an extra empty commit in #1995 and re-check.

**Comment by frelon** [comment]
solved in #1996, verified in #1997:

```
HEAD is now at 7f0b24b Merge 11806b9ef1d3ea821a2a92f2e12522c516916680 into d522f6f29ddd0cc9915bff89f7cb3b8a809e6782
```
